internal/storage/db.go

```diff
@@ -25,16 +34,339 @@ func InitDB(ctx context.Context, dsn string) (db *pgx.Conn, err error) {
 		return nil, err
 	}
 
+	// Register Numeric and decimal support
+	pgxdecimal.Register(db.TypeMap())
+
+	// Create tables for Market if not exist
+
+	query := `
+	DO $$
+	BEGIN
+		IF NOT EXISTS (SELECT 1 FROM pg_type WHERE typname = 'processing') THEN
+			CREATE TYPE processing AS ENUM ('NEW', 'PROCESSING', 'INVALID', 'PROCESSED', 'REGISTERED');
+		END IF;
+	END$$
+	`
+	_, err = db.Exec(ctx, query)
+	if err != nil {
+		return nil, fmt.Errorf("error create processing enum:  %w", err)
+	}
+
+	query = `
+	CREATE TABLE IF NOT EXISTS users (
+		id SERIAL, 
+		user_id UUID NOT NULL UNIQUE, 
+		login TEXT NOT NULL UNIQUE, 
+		password TEXT NOT NULL
+		);
+		`
+
+	_, err = db.Exec(ctx, query)
+	if err != nil {
+		return nil, fmt.Errorf("error create users %w", err)
+	}
+
+	query = `
+	CREATE TABLE IF NOT EXISTS orders (
+		id SERIAL, 
+		user_id UUID NOT NULL REFERENCES users(user_id),
+		onumber VARCHAR(20) NOT NULL UNIQUE,
+		isPreorder BOOLEAN NOT NULL, 
+		uploaded TIMESTAMPTZ NOT NULL,
+		status processing NOT NULL DEFAULT 'NEW'
+		);
+		`
+	_, err = db.Exec(ctx, query)
+	if err != nil {
+		return nil, fmt.Errorf("error create orders %w", err)
+	}
+
+	query = `
+	CREATE TABLE IF NOT EXISTS bonuses (
+		id SERIAL, 
+		onumber VARCHAR(20) NOT NULL REFERENCES orders(onumber) UNIQUE,
+		bonus_used NUMERIC DEFAULT 0,
+		bonus_accrual NUMERIC DEFAULT 0
+	);
+	`
+	_, err = db.Exec(ctx, query)
+	if err != nil {
+		return nil, fmt.Errorf("error create bonuses %w", err)
+	}
+
 	return
 }
 
-func (base *Repo) Start(ctx context.Context) error {
+func (base *RepoMarket) Start(ctx context.Context) error {
 	ctx, cancel := context.WithTimeout(ctx, time.Second*3)
 	err := base.master.Ping(ctx)
 	defer cancel()
 	return err
 }
 
-func (base *Repo) GetOrder() {
+func (base *RepoMarket) Register(ctx context.Context, user model.User) error {
```

Comment:
для базы данных не важен что контекст это регистрация (не факт что именно так будет использоваться этот метод, может быть это выгрузка из эксель со старого сайта)

```suggestion
func (base *RepoMarket) AddUser(ctx context.Context, user model.User) error {
```

Reply:
Понял, все методы относятся с storage и не тянут логику других слоев, исправил.